src/v/cloud_storage/inventory/aws_ops.cc

```diff
@@ -87,12 +89,26 @@ aws_ops::create_inventory_configuration(
       .prefix = _prefix,
       .frequency = frequency};
 
-    const auto key = fmt::format("?inventory&id={}", _inventory_config_id());
     co_return co_await remote.upload_object(
       {.transfer_details
-       = {.bucket = _bucket, .key = cloud_storage_clients::object_key{key}, .parent_rtc = parent_rtc},
+       = {.bucket = _bucket, .key = _inventory_key, .parent_rtc = parent_rtc},
        .type = upload_type::inventory_configuration,
        .payload = to_xml(cfg)});
 }
 
+ss::future<bool> aws_ops::inventory_configuration_exists(
+  cloud_storage::cloud_storage_api& remote, retry_chain_node& parent_rtc) {
+    // This buffer is thrown away after the GET call returns, we might introduce
+    // some sort of struct (or reuse the aws_report_configuration used for
+    // serialization), but for AWS we generally do not need the actual report
+    // config, the HTTP status for the GET call is sufficient to tell us if the
+    // config exists.
+    iobuf buf;
+    auto dl_res = co_await remote.download_object(
+      {.transfer_details
+       = {.bucket = _bucket, .key = _inventory_key, .parent_rtc = parent_rtc},
+       .payload = buf});
```

Comment:
is it slightly odd to use download_object here? seems like some sort of more raw/low-level GET verb used directly would be clearer? (btw no need to chagne this , just trying to understand)

Reply:
It is a little off, but `remote` currently doesn't have a pure GET operation, `download_object` was also added recently. The right thing to do would probably be to add a GET operation and make `download_object` a wrapper for it.